### DISCUSSION TITLE
No internet

### DISCUSSION THREAD

**Discussion by lucmp** [SpiralLinux/SpiralLinux-project#292] [Q&A] [answered]
I finally installed Spiral on bare metal. And I have no internet.

I have an Ethernet connection so I thought it should work right out of the box. Well, it seems my modem also requires a password for the Ethernet connection. But I'm not sure. I would have to try it.

But how? There is no connection manager.

I don't know what to do.

**Comment by geckolinux** [comment]
Hi there, we would need to know what kind of ethernet device you have. If you don't know you can run:
`inxi -N`

**Reply by lucmp** [reply]
`Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller driver: r8169`

**Reply by ionmich1** [reply]
Same problem Realtek RTL8111/8168/8411 r8168

**Reply by geckolinux** [reply]
@lucmp and @ionmich1 Please try uninstalling the `r8168-dkms` package and then reboot.

**Reply by lucmp** [reply]
I think it got worse. At least eth0 was shown in the output of 'ip a'. It is not anymore.

Of course, internet is still not working.

I copied the log of the removal in synaptic to a text file, opened it and it looked good. I saved it in a pen drive. Now I open the file in another Debian install and everything is very very garbled. Like it's encrypted or something.

**Reply by lucmp** [reply]
I copied the file again. This time it worked.

```
(Reading database ... 106174 files and directories currently installed.)
Removing r8168-dkms (8.050.03-1~bpo11+1) ...

-------- Uninstall Beginning --------
Module:  r8168
Version: 8.050.03
Kernel:  5.10.0-18-amd64 (x86_64)
-------------------------------------

Status: Before uninstall, this module version was ACTIVE on this kernel.

r8168.ko:
 - Uninstallation
   - Deleting from: /lib/modules/5.10.0-18-amd64/updates/dkms/
 - Original module
   - No original module was found for this module on this kernel.
   - Use the dkms install command to reinstall any previous module version.

depmod....

DKMS: uninstall completed.

-------- Uninstall Beginning --------
Module:  r8168
Version: 8.050.03
Kernel:  5.18.0-0.deb11.4-amd64 (x86_64)
-------------------------------------

Status: Before uninstall, this module version was ACTIVE on this kernel.

r8168.ko:
 - Uninstallation
   - Deleting from: /lib/modules/5.18.0-0.deb11.4-amd64/updates/dkms/
 - Original module
   - No original module was found for this module on this kernel.
   - Use the dkms install command to reinstall any previous module version.

depmod....

DKMS: uninstall completed.

------------------------------
Deleting module version: 8.050.03
completely from the DKMS tree.
------------------------------
Done.
Processing triggers for initramfs-tools (0.140) ...
update-initramfs: Generating /boot/initrd.img-5.18.0-0.deb11.4-amd64
W: Possible missing firmware /lib/firmware/i915/tgl_huc_7.9.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/tgl_huc_7.9.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/dg1_huc_7.9.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/tgl_huc_7.9.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/tgl_huc_7.9.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/skl_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/bxt_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/kbl_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/glk_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/kbl_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/kbl_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/cml_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/icl_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/ehl_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/ehl_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/tgl_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/tgl_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/dg1_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/tgl_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/adlp_guc_69.0.3.bin for module i915
W: Possible missing firmware /lib/firmware/i915/tgl_dmc_ver2_12.bin for module i915
W: Possible missing firmware /lib/firmware/i915/rkl_dmc_ver2_03.bin for module i915
W: Possible missing firmware /lib/firmware/i915/adlp_dmc_ver2_14.bin for module i915
I: The initramfs will attempt to resume from /dev/dm-1
I: (/dev/mapper/luks-[uuid])
I: Set the RESUME variable to override this.

```

**Reply by geckolinux** [reply]
@lucmp Does your ethernet work with the vanilla Debian installation?

**Reply by lucmp** [reply]
I am on Debian 9 now.
Yes, it's been working for years, since I installed it. I also had 8 on this machine before 9.
I never tried any newer Debian except for Spiral.

**Reply by geckolinux** [reply]
@lucmp And did you try the 5.10 and the 5.18 and the 6.x kernels?

**Reply by lucmp** [reply]
I almost forgot, I've had this output in synaptic for about a year or two every time I install or uninstall something. The installs work though. So does internet.

```
Processing triggers for gnome-menus (3.13.3-9) ...
Processing triggers for hicolor-icon-theme (0.15-1) ...
Processing triggers for initramfs-tools (0.130) ...
update-initramfs: Generating /boot/initrd.img-4.9.0-3-amd64
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8107e-2.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8107e-1.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168h-2.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168h-1.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168g-3.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168g-2.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8106e-2.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8106e-1.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8411-2.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8411-1.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8402-1.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168f-2.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168f-1.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8105e-1.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168e-3.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168e-2.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168e-1.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168d-2.fw for module r8169
W: Possible missing firmware /lib/firmware/rtl_nic/rtl8168d-1.fw for module r8169
cryptsetup: WARNING: failed to detect canonical device of /dev/sda1
cryptsetup: WARNING: could not determine root device from /etc/fstab
Fatal: Trying to map files from unnamed device 0x0014 (NFS/RAID mirror down ?)
run-parts: /etc/initramfs/post-update.d//runlilo exited with return code 1
dpkg: error processing package initramfs-tools (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 initramfs-tools

Not all changes and updates succeeded. For further details of the failure, please expand the 'Details' panel below.


W: Download is performed unsandboxed as root as file '/root/.synaptic/tmp//tmp_sh' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)
```

**Reply by lucmp** [reply]
I only tried Spiral with the 5.18 kernel.

**Reply by lucmp** [reply]
Just so you know, I rolled back a snapshot to before I removed the r8168-dkms package and booted with kernel 5.10.

No difference. All the same.

One thing I find very strange is that `ip a` shows lo, eth0 and wlan0, but when I try `ifup eth0` the output is `unknown interface eth0` (or wlan0, respectively).

**Reply by geckolinux** [reply]
@lucmp 
> before I removed the r8168-dkms package and booted with kernel 5.10

Please try kernel 5.10 after without the `r8168-dkms` package.

**Reply by lucmp** [reply]
I tried that too. No avail either.

I managed to connect to wifi with nmcli and could install packages so the kernel and modules work. No real need to remove r8168-dkms.

Maybe the problem is the password requirement. I don't know how to connect to Ethernet with a password. I found that wicd and wicd-gtk were removed from Debian. So I tried an abomination called connman and it didn't work. So my problem now is to find a connection manager that doesn't look and behave like something worse than even Microsoft could possibly unleash upon mankind.

Conclusion: the internet works. It's just become exceptionally difficult to use.

Thank you once again.

**Reply by ionmich1** [reply]
After installation and boot from hard drive if I don't upgrade immediately eth0 works. There is a r8169 module. After I upgrade it stops working.

**Reply by geckolinux** [reply]
@ionmich1 There is a known bug in the latest Debian Bullseye Backports kernel version. Please confirm if the standard 5.10 kernel does everything you need, and if so revert to that.

**Reply by ionmich1** [reply]
I will check that. Meanwhile...When I boot from my hard drive after installing eth0 works just fine. After I do an apt upgrade it fails.

**Reply by ionmich1** [reply]
`ion@ion-ms7c52:~$ uname -r
5.18.0-0.deb11.4-amd64
ion@ion-ms7c52:~$ 
`
And I have a connection as I write this.

**Reply by geckolinux** [reply]
@ionmich1 That's correct, the bug did not exist in the default Debian 5.10 kernel or in the Backports 5.18 kernel that was the the latest at the time of the SpiralLinux ISO creation. Later it was upgraded to 6.x in Backports which is what you get after upgrading the installation, and that's when the bug started. I would recommend that you:
- Option 1: Remove `/etc/apt/preferences.d/99backports` and then uninstall all kernels except 5.10.x.
- Option 2: Upgrade to Debian 12 https://github.com/SpiralLinux/SpiralLinux-project/wiki#upgrading-to-a-major-new-debian-stable-release

**Reply by lucmp** [reply]
@ionmich1 If you upgrade to Debian 12, please tell us what happens.

**Reply by ionmich1** [reply]
Nothing changed. After a week I started from a new install and updated to Debian 12. Still no ethernet. But if I boot 5.18.16-1~bpo11+1 it works.

**Reply by ionmich1** [reply]
5.18.16-1 loads module r8168 and it connects. Debian 12 does not include this module or any other rxxxx moduloe. So how do I copy r8168 from the 5.18.16-1 boot, add it permanently to the Debian 12 kernel version and insure that the next update doesn't trash my efforts?

**Reply by lucmp** [reply, marked as answer]
Maybe this should be reported upstream. I mean, to Debian directly.
